src/k2/dto/FieldTypes.cpp

```diff
@@ -45,8 +45,14 @@ template <> FieldType TToFieldType<int16_t>() { return FieldType::INT16T; }
 template <> FieldType TToFieldType<int32_t>() { return FieldType::INT32T; }
 template <> FieldType TToFieldType<int64_t>() { return FieldType::INT64T; }
 template <> FieldType TToFieldType<float>() { return FieldType::FLOAT; }
+template <> FieldType TToFieldType<double>() { return FieldType::DOUBLE; }
+template <> FieldType TToFieldType<bool>() { return FieldType::BOOL; }
+template <> FieldType TToFieldType<FieldType>() { return FieldType::FIELD_TYPE; }
 
 // All conversion assume ascending ordering
+template <> String FieldToKeyString<bool>(const bool& field) {
+    return field? "1": "0";
```

Comment:
You need the type byte and terminator bytes here as well, or you can remove this function if you don't want to support bool fields as keys

Reply:
removed